FhirToDataLake/docs/Process FHIR extensions.md

```diff
@@ -0,0 +1,101 @@
+# Process FHIR extensions  
+
+ ![image](./assets/ProcessExtensions.png)
+
+## Generate parquet data for FHIR extensions
+
+### 1.	Prepare the customized schema templates
+For each resource types, e.g. Patient, Observation, you need to prepare a **liquid template** and a related **JSON schema** file.
+- The liquid template will be used to convert the raw FHIR JSON data into target structure.
+- The JSON schema will be used to generate the Parquet schema and create table definitions.
+
+Follow the links for more information about [liquid template](http://dotliquidmarkup.org/) and [JSON schema](https://json-schema.org/learn/getting-started-step-by-step).
+
+Example liquid template and JSON schema for “Patient” resource, flatten the “birthplace” extension for analytics.
+
+_Liquid template:_
+```liquid
+{% validate "Schema/Patient.schema.json" -%}
+{
+"resourceType": "{{ msg.resourceType }}",
+"id": "{{ msg.id }}",
+
+{% assign birth_place_extension = msg["extension"] | where: "url", "http://hl7.org/fhir/StructureDefinition/patient-birthPlace" | first -%}
+"birthPlaceExtension.valueAddress.city": "{{ birth_place_extension.valueAddress.city }}",
+"birthPlaceExtension.valueAddress.state": "{{ birth_place_extension.valueAddress.state }}",
+"birthPlaceExtension.valueAddress.country": "{{ birth_place_extension.valueAddress.country }}",
+"birthPlaceExtension.valueAddress.postalCode": "{{ birth_place_extension.valueAddress.postalCode }}",
+}
+{% endvalidate -%}
+
+```
+
+_JSON schema file:_
+
+```javascript
+{
+    "title": "Patient customized schema",
+    "type": "object",
+    "properties": {
+        "resourceType": { type": "string" },
+        "id": { "type": "string" },
+	"birthPlaceExtension.valueAddress.city": { type": "string" },
+	"birthPlaceExtension.valueAddress.state": { "type": "string" },
+	"birthPlaceExtension.valueAddress.country": { "type": "string" },
+	"birthPlaceExtension.valueAddress.postalCode": { "type": "string" }
+    },
+    "required": [ "id" ]
+}
+```
+
+**Note**:
+1.	The JSON schema files must be saved at **Schema** directory in the image.
+	 
+	 ![image](./assets/LiquidDirectory.png)
+	 
```

Comment:
I missed a requirement, the JSON schema files need to have suffix **".schema.json"**